iina/PlayerCore.swift

```diff
@@ -667,30 +667,40 @@ class PlayerCore: NSObject {
     syncUITime()
   }
 
-  func setCrop(fromString str: String) {
+  func setCrop(fromString str: String) -> Bool? {
     let vwidth = info.videoWidth!
     let vheight = info.videoHeight!
     if let aspect = Aspect(string: str) {
       let cropped = NSMakeSize(CGFloat(vwidth), CGFloat(vheight)).crop(withAspect: aspect)
       let vf = MPVFilter.crop(w: Int(cropped.width), h: Int(cropped.height), x: nil, y: nil)
       vf.label = Constants.FilterName.crop
-      setCrop(fromFilter: vf)
+      if !setCrop(fromFilter: vf) {
+        return false
+      }
       // warning! may should not update it here
       info.unsureCrop = str
       info.cropFilter = vf
+      return true
     } else {
       if let filter = info.cropFilter {
         let _ = removeVideoFilter(filter)
         info.unsureCrop = "None"
       }
     }
+    return nil
   }
 
-  func setCrop(fromFilter filter: MPVFilter) {
+  func setCrop(fromFilter filter: MPVFilter) -> Bool {
```

Comment:
You may also want to consider making this `@discardableResult` if you plan on ignoring the return value often.